### PR TITLE
Google authentication, session management, initial routing - AB#42

### PR DESCRIPTION
**Pull request by Kesuera**
I have added google authentication and session management, link to library: https://github.com/sidebase/nuxt-auth. At the same time, some primitive routing was added. Part about our team - non-authorized, part for the project - authorized.

### CODE REVIEW

File: pages/index.vue [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div>
+    <h1>Home page</h1>
+    <button @click="goToPage('/team/about')">Team section</button>
+    <button @click="goToPage('/project/login')">Project section</button>
```
Comment (bran0h) [inline pages/index.vue]: Router instance is auto-injected in nuxt, you can use `$router.push(...)` instead of custom function

File: layouts/team.vue
Comment (bran0h) [inline layouts/team.vue:1]: I don't think this layout is necessary, we can use default layout for team instead

File: pages/index.vue [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  </div>
+</template>
+
+<script lang="ts" setup>
+definePageMeta({
+  auth: false,
```
Comment (bran0h) [inline pages/index.vue:10]: I would move this meta to default layout

File: pages/team/about.vue [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<template>
+  <div>
+    <h1>About page</h1>
+  </div>
+</template>
+
+<script lang="ts" setup>
+definePageMeta({
+  auth: false,
+  layout: "team",
```
Comment (bran0h) [inline pages/team/about.vue]: As mentioned earlier

**Review by bran0h — CHANGES_REQUESTED** [review]
Please add `.env.example` file, so we know which variables should be provided

**Comment by Kesuera** [comment]
I added a .env.example file, changed routing to use $router.push(...) and made some layout changes => default layout is for project, team layout is preserved and default index page is the team dashboard page from which user can move to project section.

**Review by bran0h — DISMISSED** [review]

**Review by steboee — APPROVED** [review]

**Pull request merged**